onnx/backend/test/runner/__init__.py

```diff
@@ -145,13 +145,22 @@ def _prepare_model_data(self, model_test):
             os.makedirs(model_dir)
             url = 'https://s3.amazonaws.com/download.onnx/models/{}.tar.gz'.format(
                 model_test.model_name)
-            with tempfile.NamedTemporaryFile(delete=True) as download_file:
+
+            # On Windows, NamedTemporaryFile can not be opened for a
+            # second time
+            download_file = tempfile.NamedTemporaryFile(delete=False)
+            try:
+                download_file.close()
                 print('Start downloading model {} from {}'.format(
                     model_test.model_name, url))
                 urlretrieve(url, download_file.name)
                 print('Done')
                 with tarfile.open(download_file.name) as t:
                     t.extractall(models_dir)
+            finally:
```

Comment:
shouldn't this be an `except` block rather than a `finally` block? @dzhulgakov @bddppq

Reply:
`NamedTemporaryFile(delete=True)` (the original behavior) was going to delete the temp file no matter the download succeeded or not.

Reply:
sure, but it wasn't going to lie about the test failing

Reply:
Oh the print line, I was thinking about the `os.remove` line. Yes you are right, that should belong to `except` block